homeassistant/components/ptdevices/sensor.py

```diff
@@ -0,0 +1,261 @@
+"""Sensors for PTDevices device."""
+
+from __future__ import annotations
+
+import logging
+from string import ascii_letters
+from typing import Any
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    PERCENTAGE,
+    SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
+    EntityCategory,
+    UnitOfLength,
+    UnitOfTemperature,
+    UnitOfVolume,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import PTDevicesCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+RESOURCES: dict[str, SensorEntityDescription] = {
+    "percent_level": SensorEntityDescription(
+        key="percent_level",
```

Comment:
You could store these keys is a StrEnum instead.

EG: https://github.com/home-assistant/core/blob/2ae23b920ad25733e1346b7993442957d22406b5/homeassistant/components/nintendo_parental_controls/sensor.py#L26